src/shared/sleep-config.c

```diff
@@ -390,19 +417,31 @@ int find_hibernate_location(HibernateLocation **ret_hibernate_location) {
                 }
         }
 
-        if (!hibernate_location)
-                return log_debug_errno(SYNTHETIC_ERRNO(ENOSYS), "No swap partitions or files were found");
+        bool resume_match = location_is_resume_device(hibernate_location, sys_resume, sys_offset);
+
+        /* resume= is set, but a matching /proc/swaps entry was not found */
+        if (!resume_match && sys_resume != 0) {
+                log_debug("/sys/power/resume appears to be configured but a matching swap in /proc/swaps could not be identified; hibernation may fail");
```

Comment:
This is still not OK. We *can* distinguish the cases where:
- there is no swap in /proc/swaps
- there is a device in /proc/swaps but a different device is configured in /sys/power/resume
- /sys/power/resume is populated and there's a matching device in /proc/swaps but we cannot verify the offset.

With your patch, after `swapoff -a` I get:
```
/sys/power/resume appears to be configured but a matching swap in /proc/swaps could not be identified; hibernation may fail
Unable to determine remaining swap space; hibernation may fail
Hibernation configured and possible: yes
```
But hibernation will surely fail, so saying that it "might fail" is wrong.

Reply:
See #14511. I think it DTRT.

Reply:
Thanks @keszybz - sorry you had to go make some final changes :-(

Reply:
@zachsmith thank you for working on this. It's not an easy piece of code, as all the back-and-forth shows.